src/services/session.service.ts

```diff
@@ -0,0 +1,38 @@
+import Session from "../session/session";
+import { IPromiseToFileSession } from "session/types";
+import { saveSession } from "./redis.service";
+import * as keys from "../session/keys";
+
+/**
+ * Creates a new ptf session and adds a company to the context.
+ * This will override any ptf session that is already in redis.
+ * @param chSession
+ * @param companyNumber
+ */
+const createPromiseToFileSession = async (
+    chSession: Session, companyNumber: string): Promise<IPromiseToFileSession> => {
+
+    const existing = chSession.data[keys.PTF_SESSION];
+    if (existing) {
+        return existing;
+    }
+
+    const ptfSession: IPromiseToFileSession = {
+        company_number_in_context: companyNumber,
+        ptf_requests: [],
+    };
+    chSession.appendData(keys.PTF_SESSION, ptfSession);
+    await saveSession(chSession);
+    return ptfSession;
+};
+
+/**
+ * Returns the company in context. That is the most recent company that was
+ * input in the company number screen.
+ * @param chSession
+ */
+const getCompanyNumberInContext = (chSession: Session): string => {
+    return chSession.data.ptf_session.company_in_context;
```

Comment:
COMPANY_IN_CONTEXT is in the keys.ts file - should this be used here instead? Also, why is it getting a company and not a number (or is this another naming issue?).

Reply:
Name changed